Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTipBackgroundCorners.cs

```diff
@@ -0,0 +1,130 @@
+        [SerializeField]
+        private Transform cornerBotLeft = null;
+        [SerializeField]
+        [Range(0.01f, 2f)]
+        private float cornerScale = 1f;
+        [SerializeField]
+        private ScaleModeEnum scaleMode = ScaleModeEnum.World;
+        
+        protected override void ContentChange()
+        {
+            ScaleToFitContent();
+        }
+
+        protected override void ScaleToFitContent()
+        {
+            // Get the local size of the content - this is the scale of the text under the content parent
+            Vector3 localContentSize = ToolTipContent.LocalContentSize;
+            localContentSize.z = 1;
+            // Multiply it by 0.5 to get extents
+            localContentSize *= 0.5f;
+            Vector3 localContentOffset = ToolTipContent.LocalContentOffset;
+            // Put the corner objects at the corners
+            Vector3 topLeft = new Vector3(-localContentSize.x + localContentOffset.x, localContentSize.y + localContentOffset.y, localContentOffset.x);
+            Vector3 topRight = new Vector3(localContentSize.x + localContentOffset.x, localContentSize.y + localContentOffset.y, localContentOffset.x);
+            Vector3 botRight = new Vector3(localContentSize.x + localContentOffset.x, -localContentSize.y + localContentOffset.y, localContentOffset.x);
+            Vector3 botLeft = new Vector3(-localContentSize.x + localContentOffset.x, -localContentSize.y + localContentOffset.y, localContentOffset.x);
+            if (cornerTopLeft != null)
+            {
+                cornerTopLeft.localPosition = topLeft;
+            }
+
+            if (cornerTopRight != null)
+            {
+                cornerTopRight.localPosition = topRight;
+            }
+
+            if (cornerBotRight != null)
+            {
+                cornerBotRight.localPosition = botRight;
+            }
+
+            if (cornerBotLeft != null)
+            {
+                cornerBotLeft.localPosition = botLeft;
+            }
+
+            // Now set the corner object scale based on preference
+            Vector3 globalScale = Vector3.one;
+            switch (scaleMode)
+            {
+                case ScaleModeEnum.World:
+                    Vector3 lossyScale = ToolTipContent.ContentParentTransform.lossyScale;
+                    globalScale.x /= lossyScale.x;
+                    globalScale.y /= lossyScale.y;
+                    globalScale.z /= lossyScale.z;
+                    break;
+
+                case ScaleModeEnum.Local:
+                    Vector3 localScale = cornerTopLeft.lossyScale;
+                    float smallestDimension = Mathf.Min(Mathf.Min(globalScale.x, globalScale.y), globalScale.z);
+                    globalScale = Vector3.one * smallestDimension;
+                    break;
+
+                default:
+                    throw new System.ArgumentOutOfRangeException("ScaleMode not set to valid enum value.");
```

Comment:
let's try to avoid throwing in MRTK. the exceptions will likely be handled automatically by Unity and may case a performance issue.  recommend using Debug.LogError messages instead to alert developers of an issue,

Reply:
Agreed. I already changed it to a debug error on the PR against his branch.